internal/mode/static/handler.go

```diff
@@ -131,7 +125,55 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 		}
 	}
 
-	h.cfg.statusUpdater.Update(ctx, buildGatewayAPIStatuses(graph, nginxReloadRes))
+	gwAddresses, err := getGatewayAddresses(ctx, h.cfg.k8sClient, nil, h.cfg.gatewayPodConfig)
```

Comment:
would it simplify if remove getGatewayAddresses from here completely?  and leave:

```
	h.cfg.statusUpdater.Update(ctx, buildGatewayAPIStatuses(graph, nginxReloadRes))
```

However, in status updater, store the gatewayAddresses:

```
type UpdaterImpl struct {
        gwAddresses ...
	lastStatuses lastStatuses
	cfg          UpdaterConfig
	isLeader     bool

	lock sync.Mutex
}

. . . 
func (upd *UpdaterImpl) updateGatewayAPI(ctx context.Context, statuses GatewayAPIStatuses) {
  . . . 
  		upd.writeStatuses(ctx, nsname, &v1beta1.Gateway{}, func(object client.Object) {
			gw := object.(*v1beta1.Gateway)
			gw.Status = prepareGatewayStatus(gs, upd.cfg.Clock.Now(), gwAddresses)
		})
. . .
```

This way, StatusUpdater will get the gwAddresses from h.cfg.statusUpdater.UpdateAddresses(ctx, gwAddresses) calls, and will always set GatewayAddresses to that value.

Reply:
UpdateAddresses will only happen if the Service changes. We still need the addresses on normal Gateway resource changes, which is why it's called where it is and given to the status builder.

Reply:
Correct.